File: frontend/src/types/holding-exit-signal.ts
export type HoldingExitSignal = {
  generated_at: string;
  holding_id: number;
  ticker: string;
  asset_name: string;
  available: boolean;
  action: "继续持有" | "持有观察" | "减仓观察" | "保护利润" | "纪律止损";
  confidence: number;
  summary: string;
  thesis: string;
  evidence: string[];
  disagreement: string[];
  invalid_conditions: string[];
  risk_controls: string[];
  profit_protection_view: string;
  time_horizon: string;
  context_snapshot: Record<string, unknown>;
  empty_message: string | null;
};

export type HoldingExitSignalList = {
  generated_at: string;
  items: HoldingExitSignal[];
  count: number;
};

File: python/valuecell/server/api/schemas/holding_exit_signal.py
from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class HoldingExitSignalData(BaseModel):
    generated_at: datetime
    holding_id: int
    ticker: str
    asset_name: str
    available: bool
    action: str
    confidence: int
    summary: str
    thesis: str
    evidence: list[str] = Field(default_factory=list)
    disagreement: list[str] = Field(default_factory=list)
    invalid_conditions: list[str] = Field(default_factory=list)
    risk_controls: list[str] = Field(default_factory=list)
    profit_protection_view: str
    time_horizon: str
    context_snapshot: dict[str, Any]
    empty_message: str | None = None


class HoldingExitSignalListData(BaseModel):
    generated_at: datetime
    items: list[HoldingExitSignalData] = Field(default_factory=list)
    count: int
